linguistics/src/main/java/com/yahoo/language/simple/SimpleLinguistics.java

```diff
@@ -19,15 +20,35 @@
  * Factory of pure Java linguistic processor implementations.
  *
  * @author bratseth
+ * @author bjorncs
  */
 public class SimpleLinguistics implements Linguistics {
 
     // Threadsafe instances
-    private final static Normalizer normalizer = new SimpleNormalizer();
-    private final static Transformer transformer = new SimpleTransformer();
-    private final static Detector detector = new SimpleDetector();
-    private final static CharacterClasses characterClasses = new CharacterClasses();
-    private final static GramSplitter gramSplitter = new GramSplitter(characterClasses);
+    private final Normalizer normalizer;
+    private final Transformer transformer;
+    private final Detector detector;
+    private final CharacterClasses characterClasses;
+    private final GramSplitter gramSplitter;
+
+    @Inject
+    public SimpleLinguistics() {
+        CharacterClasses characterClasses = new CharacterClasses();
+        this.normalizer = new SimpleNormalizer();
+        this.transformer = new SimpleTransformer();
+        this.detector = new SimpleDetector();
+        this.characterClasses = new CharacterClasses();
+        this.gramSplitter = new GramSplitter(characterClasses);
+    }
+
+    public SimpleLinguistics(SimpleLinguisticsConfig config) {
+        CharacterClasses characterClasses = new CharacterClasses();
```

Comment:
Unnecessary code duplication

Reply:
The `GramSplitter` takes a `CharacterClasses` instance, and I only want a single instance of the latter type. That's why there is code duplication, otherwise I had to resolve to static factory methods.